modules/messages/src/weights_ext.rs

```diff
@@ -122,6 +145,146 @@ pub fn ensure_able_to_receive_confirmation<W: WeightInfoExt>(
+
+	ensure_weight_components_are_not_zero(weight_when_proof_size_is_8k);
+	ensure_weight_components_are_not_zero(weight_when_proof_size_is_16k);
+	ensure_proof_size_is_the_same(
+		weight_when_proof_size_is_8k,
+		weight_when_proof_size_is_16k,
+		"Messages proof size does not affect values that we read from our storage",
+	);
+	ensure_ref_time_lte(
+		weight_when_proof_size_is_8k,
+		weight_when_proof_size_is_16k,
+		"Larger message proofs mean more hashing and iterations => computation time",
+	);
+}
+
+/// Panics if `proof_size` of message delivery call depends on the messages count.
+///
+/// In practice, it will depend on the messages count, because most probably every
+/// message will read something from db during dispatch. But this must be accounted
+/// by the `dispatch_weight`.
+fn messages_count_does_not_affect_proof_size<W: WeightInfoExt>() {
+	let messages_proof_size = PreComputedSize(8 * 1024);
+	let dispatch_weight = Weight::zero();
+	let weight_of_one_incoming_message =
+		W::receive_messages_proof_weight(&messages_proof_size, 1, dispatch_weight);
+	let weight_of_two_incoming_messages =
+		W::receive_messages_proof_weight(&messages_proof_size, 2, dispatch_weight);
+
+	ensure_weight_components_are_not_zero(weight_of_one_incoming_message);
+	ensure_weight_components_are_not_zero(weight_of_two_incoming_messages);
+	ensure_proof_size_is_the_same(
+		weight_of_one_incoming_message,
+		weight_of_two_incoming_messages,
+		"Number of same-lane incoming messages does not affect values that we read from our storage",
+	);
+	ensure_ref_time_lte(
+		weight_of_one_incoming_message,
+		weight_of_two_incoming_messages,
+		"More incoming messages mean more computation overhead",
+	);
+}
+
+/// Panics if `proof_size` of delivery confirmation call depends on the delivery proof size.
+fn messages_delivery_proof_size_does_not_affect_proof_size<W: WeightInfoExt>() {
+	let relayers_state = UnrewardedRelayersState {
+		unrewarded_relayer_entries: 1,
+		messages_in_oldest_entry: 1,
+		total_messages: 1,
+		last_delivered_nonce: 1,
+	};
+	let weight_when_proof_size_is_8k =
+		W::receive_messages_delivery_proof_weight(&PreComputedSize(8 * 1024), &relayers_state);
+	let weight_when_proof_size_is_16k =
+		W::receive_messages_delivery_proof_weight(&PreComputedSize(16 * 1024), &relayers_state);
+
+	ensure_weight_components_are_not_zero(weight_when_proof_size_is_8k);
+	ensure_weight_components_are_not_zero(weight_when_proof_size_is_16k);
+	ensure_proof_size_is_the_same(
+		weight_when_proof_size_is_8k,
+		weight_when_proof_size_is_16k,
+		"Messages delivery proof size does not affect values that we read from our storage",
+	);
+	ensure_ref_time_lte(
+		weight_when_proof_size_is_8k,
+		weight_when_proof_size_is_16k,
+		"Larger messages delivery proofs mean more hashing and iterations => computation time",
+	);
+}
+
+/// Panics if `proof_size` of delivery confirmation call depends on the number of confirmed
```

Comment:
This isn't true in general - there may be reward schemes, which read some stuff e.g. for every message or so. But our impl doesn't do that && other may just not use our test.

Feel free to argue